### DISCUSSION TITLE
Unable to install MDF package

### DISCUSSION THREAD

**Discussion by ghost** [ModECI/MDF#212] [Q&A] [answered]
Hi everyone,
On getting started with the project contributions we are asked to install the MDF package and all its dependencies and the repo 'https://github.com/ModECI/MDF' is linked. However, I cannot find any installation package except for the 'https://github.com/ModECI/MDF/blob/main/install_on_osbv2.sh' folder which is doing the imports on open-source brains ( trying to do this, I am still getting an error while installing the MDF package).

Has anyone gone past this yet?
Hi mentors @sanjayankur31 @FranciscoD @pgleeson, Please how do I go about this?

**Comment by pgleeson** [comment, marked as answer]
@gno-lit Have a look at the https://github.com/ModECI/MDF/blob/main/test_all.sh script. This outlines the sequence of install and example execution lines to run. Basically do a `pip install .` and run the tests, them run the examples.

**Reply by somyagr** [reply]
Hi @pgleeson , thanks for your help but `pip install .` is not working. I tried to run it directly after cloning the repository. The terminal closes as soon as I enter the command. Please help me through this.

**Reply by sanjayankur31** [reply]
Could you give us some more information on this please @somyagr ? What operating system are you using? What version of Python?

The steps should be:

- clone the repo
- cd into the cloned repo
- create a virtual environment: [native python](https://docs.python.org/3/tutorial/venv.html) or conda
- activate the virtual environment
- run `pip install .`

This *should* work

**Reply by FatimaArshad-DS** [reply]
Tried this on MAC. Got this Command errored out with exit status 1. Not working

**Reply by FatimaArshad-DS** [reply]
python 3.8

**Reply by somyagr** [reply]
Thanks for replying, I am using Ubuntu 20.04.1, Python 2.7.18
I am stuck in 5th step

**Comment by sanjayankur31** [comment]
Could you try with Python 3 please and see how that goes? (Python 2 is
quite old now and no longer developed/supported)

**Reply by FatimaArshad-DS** [reply]
Python 3 isn't working either

**Reply by FatimaArshad-DS** [reply]
Can you generate a requirements.txt and share it here? That might work

**Comment by FatimaArshad-DS** [comment]
I tried using python setup.py install
It is working fine except in the end it says error: protobuf 4.0.0rc2 is installed but protobuf<4.0dev,>=3.5.0.post1 is required by {'grpcio-tools'}

It is automatically installing protobuf. I tried downgrading but doesn't work. Can anyone share their protobuf version

**Reply by sanjayankur31** [reply]
Did you create a new virtual environment?

**Reply by sanjayankur31** [reply]
(I've just double checked in a fresh `python3.9` virtual environment, and it installs fine here)

**Reply by somyagr** [reply]
It worked for me, thankyou so much @sanjayankur31 
Now I can proceed with the contributions

**Reply by FatimaArshad-DS** [reply]
Still not working. Tried in Python3.9 with fresh env ERROR: Command errored out with exit status 1: /opt/anaconda3/envs/py39/bin/python /opt/anaconda3/envs/py39/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /var/folders/gh/p5hf8g250_n683gnhx3pt6pc0000gn/T/tmpikdpfli4 Check the logs for full command output.

**Reply by somyagr** [reply]
@FatimaArshad-DS 
```
pip install virtualenv
cd MDF
virtualenv MDF-env
source MDF-env/bin/activate
pip install .
```
These are the command lines I followed after cloning the project. Used pip version 22.0.4 and python verison 3.9

**Reply by FatimaArshad-DS** [reply]
I cloned the repo rather than downloading it directly and is working fine. Not sure what's the difference intuitively. Should have worked the other way too.

**Reply by FatimaArshad-DS** [reply]
Thank you @somyagr and @sanjayankur31

**Reply by pgleeson** [reply]
Glad you solved it @FatimaArshad-DS. The onnx/protobuf packages are proving tricky when other packages are installed. Note there are some specific versions of packages listed here: https://github.com/ModECI/MDF/blob/main/install_on_osbv2.sh#L23 which

**Reply by shivani6320** [reply]
Hi @pgleeson @sanjayankur31 , when I try to intsall it gives me this error:
```
ERROR: Cannot install modeci-mdf==0.1.0, modeci-mdf==0.2.0, modeci-mdf==0.3.0, modeci-mdf==0.3.1, modeci-mdf==0.3.2 and modeci-mdf==0.3.3 because these package versions have conflicting dependencies.

The conflict is caused by:
    modeci-mdf 0.3.3 depends on onnxruntime<=1.7.0
```
I'm unable to install onnxruntime:
```
ERROR: Could not find a version that satisfies the requirement onnxruntime (from versions: none)
ERROR: No matching distribution found for onnxruntime
```
Can you please help me out here?

**Reply by sanjayankur31** [reply]
Is this in a virtual environment? What version of python are you using?

**Reply by shivani6320** [reply]
yes, I created a virtual env. I'm using python version 3.8.1

**Reply by pgleeson** [reply]
@shivani6320 Have you tried explicitly installing onnxruntime with `pip install onnxruntime==1.7.0`? What OS are you on?

**Reply by shivani6320** [reply]
yes I tried, it gives me error as I mentioned above. I'm using windows 10 with 64 bit version.

**Reply by pgleeson** [reply]
@shivani6320 I've noticed that we were only testing on Windows with python 3.7, and 3.9+windows was excluded as it had known problems. I've created a branch testing more py combinations here: https://github.com/ModECI/MDF/actions/runs/2110699191.

It could be an issue that onnx<=1.7.0 (or other libraries) are not supported on win+3.8... One short term fix (apart from us loosening our dependencies: https://github.com/ModECI/MDF/issues/230) is for you to use conda for creating python envs, where you can specify that you wnat to use py 3.7, and then install as normal...

**Reply by shivani6320** [reply]
Hey @pgleeson , I started using conda & installed the package successfully. Thank you for your time :)

**Reply by pgleeson** [reply]
@shivani6320 Glad it worked for you!

**Comment by sanjayankur31** [comment]
Hrm, could you please post the complete output of the command so we can see
what the error is?
